crates/gas-price-algorithm/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    cell::RefCell,
+    cmp::{
+        max,
+        min,
+    },
+};
+
+pub struct AlgorithmV1 {
+    // DA
+    da_max_change_percent: u8,
+    min_da_price: u64,
+    da_p_value_factor: i64,
+    da_d_value_factor: i64,
+    moving_average_profit: RefCell<i64>,
+    last_profit: RefCell<i64>,
+    profit_slope: RefCell<i64>,
```

Comment:
Why do we need `RefCell`? If you need to update the internal state, then let's use `&mut`=)

Reply:
I agree. Kinda related to the `TODO` above.